Dockerfile

```diff
@@ -1,7 +1,65 @@
-# FROM gcr.io/neuromancer-seung-import/pychunkedgraph:graph-tool_dracopy
-FROM seunglab/pychunkedgraph:graph-tool_dracopy
+ARG PYTHON_VERSION=3.10
+ARG BASE_IMAGE=tiangolo/uwsgi-nginx-flask:python${PYTHON_VERSION}
+
+
+######################################################
+# Build Image - PCG dependencies
+######################################################
+FROM ${BASE_IMAGE} AS pcg-build
+ENV PATH="/root/miniconda3/bin:${PATH}"
+ENV CONDA_ENV="pychunkedgraph"
+
+# Setup Miniconda
+RUN apt-get update && apt-get install build-essential wget -y
+RUN wget \
+    https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh \
+    && mkdir /root/.conda \
+    && bash Miniconda3-latest-Linux-x86_64.sh -b \
+    && rm -f Miniconda3-latest-Linux-x86_64.sh
+
+# Install PCG dependencies - especially graph-tool
+# Note: uwsgi has trouble with pip and python3.10, so adding this with conda, too
+COPY requirements.txt .
+COPY requirements.yml .
+RUN conda env create -n ${CONDA_ENV} -f requirements.yml
+
+# Shrink conda environment into portable non-conda env
+RUN conda install conda-pack -c conda-forge
+RUN conda-pack -n ${CONDA_ENV} -o /tmp/env.tar \
+  && mkdir -p /app/venv \
+  && cd /app/venv \
+  && tar xf /tmp/env.tar \
+  && rm /tmp/env.tar
+RUN /app/venv/bin/conda-unpack
+
+
+######################################################
+# Build Image - Bigtable Emulator (without Google SDK)
+######################################################
+FROM golang:buster as bigtable-emulator-build
+RUN mkdir -p /usr/src
+WORKDIR /usr/src
+ENV GOOGLE_CLOUD_GO_VERSION bigtable/v1.12.0
+RUN apt-get update && apt-get install git -y
+RUN git clone --depth=1 --branch="$GOOGLE_CLOUD_GO_VERSION" https://github.com/googleapis/google-cloud-go.git . \
+  && cd bigtable \
+  && go install -v ./cmd/emulator
+
+
+######################################################
+# Production Image
+######################################################
+FROM ${BASE_IMAGE}
+ENV VIRTUAL_ENV=/app/venv
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+COPY --from=pcg-build /app/venv /app/venv
+COPY --from=bigtable-emulator-build /go/bin/emulator /app/venv/bin/cbtemulator
+COPY override/gcloud /app/venv/bin/gcloud
 COPY override/timeout.conf /etc/nginx/conf.d/timeout.conf
 COPY override/supervisord.conf /etc/supervisor/conf.d/supervisord.conf
-COPY requirements.txt /app
```

Comment:
this pattern improves docker build times when doing it locally considerably because the requirements don't change but code is often changing when trying to develop rapidly.. makes little difference when building in the cloud.  The pip lines can go if you've fixed the problems with out of date pip in this image.

Reply:
sorry i see you moved it up